controllers/grafanadashboard/dashboard_pipeline.go

```diff
@@ -112,7 +112,7 @@ func (r *DashboardPipelineImpl) validateJson() error {
 // 3) no configmap specified: try to use embedded json
 // 4) no json specified: try to use embedded jsonnet
 func (r *DashboardPipelineImpl) obtainJson() error {
-	if r.Dashboard.Spec.Url != "" && len(r.Dashboard.Spec.Json) == 0 {
+	if r.Dashboard.Spec.Url != "" {
 		err := r.loadDashboardFromURL()
 		if err != nil {
 			r.Logger.Error(err, "failed to request dashboard url, falling back to raw json")
```

Comment:
Do we support multiple input formats in the same dashboard? Sounds scary to me, if you have defined one dashboard type I think it might be worth to refactor this function.

If you have defined Url and it errors out just return the error.
It's not like you should have both Url and Json in the same dashboard. Or am I missing some obvious use case @HubertStefanski ?

Reply:
I think it's fine, as long as we prioritise URL we can keep the JSON as a fallback, which it was originally intended for. Users can leave it empty if they don't want to make use of that fallback option at all, but keeping it as a required field is the way to go IMHO.
We also do some other checks further down the function, so we quite literally "fall down" the priority stack URL -> ...

Reply:
can we resolve this thread in that case?

Reply:
Yes I think so